app/models/extract_fetcher.rb

```diff
@@ -41,6 +41,8 @@ def including(extract_ids)
   def extracts
     if fetch_minimal?
       exact_extracts
+    elsif fetch_additional?
+      subject_extracts + exact_extracts
```

Comment:
I might be wrong but isn't this going to end up including some extracts twice? Your changes to `ClassificationPipeline` construct the `ExtractFetcher` with `extract_ids + prior_extracts` which should be: the ids of every extract which triggered this reduction and also any extracts associated with any of their linked subjects. Then `subject_extracts` from `ExtractFetcher` should include all extracts associated with any subject in any of the ids specified when constructing the `ExtractFetcher`... which will be some novel extracts but also every extract already specified as well

Reply:
It's fine that there's no handling here for `user_extracts` because this is obviously going to only be used with reduction by subject, but I wonder if the name `:fetch_additional` could be clearer on this?

Reply:
As far as I could tell, the only time that extract ids were passed directly to the ExtractFetcher is when running reduction mode is on. Under normal circumstances, the Fetcher builds its own scope (the `user_extracts` and `subject_extracts` methods) with the filter that gets built in ClassificationPipeline#reduce. I think that `extract_ids` in the context of the reduce method is almost always an empty array. This allows that to still occur while allowing you to include additional extracts by changing the strategy.

Re: strategy name... `fetch_additional_subjects`? `fetch_subjects_and_exacts`? Suggestions welcome if it's not clear.

Reply:
Ohhhh, okay. Given that you've been in the code more recently than I have and that your tests pass, I'm going to assume you're right and I'm wrong :) Also, looking more at the code I recall that this is not a user-facing enumeration so it's not so important to be picky about it.

Reply:
I'm looking into this. I'm not sure yet if it's wrong per se, but I do want to start off with that it's hard to follow, because `exact_extracts` in the case of `fetch_minimal` means something different than when using `fetch_additional`. I think I don't like the usage of that here, and fixing that would also alleviate all concerns. I'll leave a more detailed suggestion elsewhere in the PR.